src/Controller/EnMarche/EventManagerController.php

```diff
@@ -189,4 +165,50 @@ public function contactMembersAction(Request $request, Event $event): Response
             'form' => $form->createView(),
         ]);
     }
+
+    /**
+     * @Route("/inscrits/imprimer", name="app_event_print_members")
+     * @Method("POST")
+     */
+    public function printMembersAction(Request $request, Event $event): Response
+    {
+        $registrations = $this->getRegistrations($request, $event, self::ACTION_PRINT);
+
+        if (!$registrations) {
+            return $this->redirectToRoute('app_event_members', [
+                'slug' => $event->getSlug(),
+            ]);
+        }
+
+        $html = $this->renderView('events/print_members.html.twig', [
+            'registrations' => $registrations,
+        ]);
+
+        return new PdfResponse($this->getPdfForResponse($html), 'Liste des participants.pdf');
+    }
+
+    private function getRegistrations(Request $request, Event $event, string $action): array
+    {
+        if (!$this->isCsrfTokenValid(sprintf('event.%s_members', $action), $request->request->get('token'))) {
+            throw $this->createAccessDeniedException("Invalid CSRF protection token to $action members.");
+        }
+
+        if (!$uuids = json_decode($request->request->get(sprintf('%ss', $action)), true)) {
+            if (self::ACTION_CONTACT === $action) {
```

Comment:
Should be handled by catching an `EmptyContactEventRegistrationRequestException` in the controller.

Reply:
It seems better to me to show an explicit message to user instead of throwing Exception.

Reply:
I said `catching` not `throwing` :p

Reply:
Ok got it, could you at least add a private const as array for valid actions and throw an invalid argument exception when an unsupported action is passed.